source/containers/Popup/Popup.tsx

```diff
@@ -0,0 +1,27 @@
+import React, { useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { View } from 'react-native';
+
+interface Props {
+  autoHideDuration?: number;
+  onClose: () => void;
+  renderPopup: (close: () => void) => React.ReactNode;
+}
+
+const Popup: React.FC<Props> = ({ autoHideDuration, onClose, renderPopup }) => {
```

Comment:
This component feels like a molecule component to me, however would like a defined layout for the Popup. Why do we outsource the rendering behavior? Is not all popups of the same design?

Reply:
My thinking here is that maybe they are not. Ie. as it is now we can use the same code for displaying a banner at the top or bottom, or a floating popup. 

I also ran into another issue with this, because at first I did as you suggest and had a basic layout container, into which I passed children. However, it was difficult to adjust the size of the container, and to get things positioned correctly; i.e. the floating container would not resize, and things would either get cut off or there would be a lot of blank space etc. This is very possibly due to me not knowing enough 'react native css', but it seems to not be so easy.